src/main/java/org/robotframework/remoteswinglibrary/agent/FindAppContextWithWindow.java

```diff
@@ -29,16 +32,24 @@
     String host;
     int port;
     boolean debug;
+    boolean closeSecurityDialogs;
+    RobotConnection robotConnection;
 
-    public FindAppContextWithWindow(String host, int port, boolean debug) {
+    HashMap<Dialog, SecurityDialogAccepter> dialogs = new HashMap<Dialog, SecurityDialogAccepter>();
+
+    public FindAppContextWithWindow(String host, int port, boolean debug, boolean closeSecurityDialogs) {
         this.host = host;
         this.port = port;
         this.debug = debug;
+        this.closeSecurityDialogs = closeSecurityDialogs;
     }
 
     public void run()  {
         try {
-            sun.awt.SunToolkit.invokeLaterOnAppContext(getAppContextWithWindow(), new ServerThread(host, port, debug));
+            robotConnection = new RobotConnection(host, port);
+            //robotConnection.connect();
```

Comment:
Uncomment this line (and comment other `connect`) to see 2nd problem I have mentioned.